blogs/2022-06-28-tips-kafka-streams-distributed.md

```diff
@@ -59,64 +59,64 @@ All join provided by Kafka Streams were designed thinking with aggregation in mi
 
-With all of that, we are sure that the execution state will always be the last version whatever the number of tasks results coming in parallel.
+With all the tasks above, we are sure that the execution state will always be the last version, whatever the number of tasks results coming in parallel.
 
-## Easy distributed workload between multiple backends
+## Easy distribution of workload between multiple backends
 
-In [Kestra](https://github.com/kestra-io/kestra), we have a scheduler that will look up all flows to schedule (timed-based) execution or with a long-pooling mechanism (detect file on file systems like S3 or SFTP). As we wanted to have no single point of failure also on this service, we need to split all the flow between all instances of schedulers.
+In [Kestra](https://github.com/kestra-io/kestra), we have a scheduler that will look up all flows to schedule (timed-based) execution or use a long-pooling mechanism (detect the file on file systems like S3 or SFTP). Since we don't want to have a single point of failure in this service, we need to split the entire flow between all instances of schedulers.
 
-To handle that case, we rely on Kafka's consumer groups that will handle the whole complexity of a distributed system for us. The logic is :
+To handle that case, we rely on Kafka's consumer groups, which will handle the whole complexity of a distributed system for us. The logic is:
 - Create a [Kafka stream](https://github.com/kestra-io/kestra/blob/develop/runner-kafka/src/main/java/io/kestra/runner/kafka/KafkaFlowListeners.java) that will read in a `KTable` and transmit all the result to a `Consumer`
 - Listen to state change, meaning mostly `REBALANCED` Streams, and empty all the flows for the `Consumer`
 - On the `READY` state, read all the `KTable` again.
 
-With these, all flows will be dispatched on all listeners, that's mean if you have 1,000 flows, approximately every consumer will have ~ 500 flows (depending on the repartition of keys). Kafka will handle all the heavy parts of the distributed systems :
-- Heartbeat to detect failure for a consumer
-- Notifying of rebalancing
-- Ensure exactly-once pattern for a topic, giving the assurance that only one consumer will handle the data.
+With these, all flows will be dispatched to all listeners. This means that if you have 1,000 flows, every consumer will have ~ 500 flows (depending on the repartition of keys). Kafka will handle all the heavy parts of the distributed systems:
+- Heartbeat to detect failure for a consumer.
+- Notifications relating to rebalancing.
+- Ensure exactly onc pattern for a topic, ensuring that only one consumer will handle the data.
 
-This way you will have a fully distributed system thanks to Kafka without the pain to pass a Jespen analysis.
+This way, you will have a fully distributed system, thanks to Kafka, without the pain of going through a Jespen analysis.
```

Comment:
I think a brief description of what Jespen analysis is and why going through it is a pain we should avoid will add value to this section.